vaadin-grid-flow-parent/vaadin-grid-flow-integration-tests/src/test/java/com/vaadin/flow/component/grid/it/GridTestPageIT.java

```diff
@@ -487,8 +487,15 @@ private String getInnerText(WebElement element) {
     public static Map<String, Map<String, ?>> getItems(WebDriver driver,
             WebElement element) {
         Object result = ((JavascriptExecutor) driver).executeScript(
-                "const items = arguments[0]._dataProviderController.rootCache.items;"
-                        + "return items.reduce((obj, item, i) => ({ ...obj, [i]: item }), {});",
+                """
+                        const { items } = arguments[0]._dataProviderController.rootCache;
+                        return items.reduce((obj, item, i) => {
```

Comment:
The earlier version assumed that `gridConnector` removed items using the `delete` operator in `$connector.clear`. This created "holes" in the array, which caused `reduce` to skip over the deleted entries. This behavior is unsafe and can lead to unexpected results. I've changed both the test and implementation to clear items by setting them to `undefined`.